packages/open-truss/src/configuration/engine-v1/renderer.tsx

```diff
@@ -0,0 +1,109 @@
+import { type YamlType } from '../../utils/yaml'
+import {
+  type FrameV1,
+  type DataV1,
+  type ViewPropsV1,
+  type WorkflowV1,
+} from './config-schemas'
+import {
+  type ReactTree,
+  type COMPONENTS,
+  type OpenTrussComponent,
+} from '../apply'
+import { hasDefaultExport } from './component'
+import React from 'react'
+import DataProvider from './DataProvider'
+import { type GlobalContext } from './engine'
+
+interface FrameContext {
+  frame: FrameV1
+  globalContext: GlobalContext
+  renderFrames: (frames: FrameV1[]) => ReactTree
+  i: number
+}
+
+export function renderFrame(
```

Comment:
First, this refactor is dope and made things much more readable.

Due to its readability, I now see a further enhancement we can make -- should `renderFrame` be a `<Frame />` component? It's a function that returns JSX, aka a component, so I think yes. And given that, I wonder if the whole rendering pipeline is actually just components, as what they do is render things and in React that's typically done by components.

I wonder if what this would give OT is a an export like `RenderConfig` that takes in a config prop and does all the stuff the engine is doing like checking `version` to see how to parse the config, etc. That could be nice because right now we're doing that lke this in the demo app and it's not beautiful:

```jsx
export default function RenderConfig({
  config,
}: {
  config: string
}): JSX.Element {
  const configurationFunction = React.useMemo(
    () => applyConfiguration(components),
    [],
  )

  const renderedComponents = React.useMemo(() => {
    if (config) {
      const parsedConfig = parseYaml(config)
      return configurationFunction(parsedConfig, {})
    }
  }, [config])

  return <>{renderedComponents}</>
}
```

I'd be happy to try to do this conversion; perhaps this weekend.

Reply:
> should renderFrame be a <Frame /> component?

Conceptually that makes sense to me and allows us to hide away the implementation detail bits like you referenced above. Let's give it a try!

> I'd be happy to try to do this conversion; perhaps this weekend.

Sounds good! I'll clean up this PR / fix the bugs. How do we feel about shipping this as is and you can do the conversion in a follow up PR?

After this PR, I'll continue build out features in the engine.

Reply:
I already did it 😊 https://github.com/open-truss/open-truss/pull/95